### PR TITLE
[BEAM-574] Remove log when new records have not been read yet (KafkaIO)

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by iemejia** [comment]
The previously existing log did not seem really useful, but feel free to comment if you think this PR should be addressed in a different way.

**Comment by iemejia** [comment]
R: @rangadi @dhalperi

**Comment by rangadi** [comment]
Did you start with an idle topic? Just curious, how often do you see this in your app?
I am OK removing it.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java [hunk 1]
```diff
@@ -1076,7 +1076,6 @@ private void updateLatestOffsets() {
     @Override
     public Instant getWatermark() {
       if (curRecord == null) {
-        LOG.warn("{}: getWatermark() : no records have been read yet.", name);
```
Comment (rangadi) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java]: May be good to make it a debug log. would serve as a comment.

**Comment by iemejia** [comment]
The test is really simple, I started an empty kafka without nothing (default config for both zookeeper/kafka, and started a pipeline to read data from a topic in it.

```java
        PCollection<String> data = pipeline
            .apply("ReadFromKafka", KafkaIO.read()
                .withBootstrapServers(options.getKafkaServer())
                .withTopics(Arrays.asList(options.getKafkaTopic()))
                .withKeyCoder(StringUtf8Coder.of())
                .withValueCoder(StringUtf8Coder.of())
            )
            .apply("ExtractPayload", ParDo.of(new DoFn<KafkaRecord, String>() {
                @ProcessElement
                public void processElement(ProcessContext c) throws Exception {
                    c.output(c.element().getKV().getValue().toString());
                }
            }));
```

**Comment by iemejia** [comment]
and then I had the log message printing some ConsumerConfig and then it printed infinetely:
...
[pool-1-thread-2] WARN org.apache.beam.sdk.io.kafka.KafkaIO - Reader-0: getWatermark() : no records have been read yet.
[pool-1-thread-4] WARN org.apache.beam.sdk.io.kafka.KafkaIO - Reader-0: getWatermark() : no records have been read yet. 
...

I am not sure that keeping the log message  is that useful (even if not receiving data could be seen as an exceptional situation in kafka, because we can change the log to the trace level but then the question is how useful this is, (printing a message saying no data received for the moment) indefinitely.

**Comment by rangadi** [comment]
Agreed. We can remove the log.

**Comment by rangadi** [comment]
:+1:

**Comment by dhalperi** [comment]
LGTM, merged. Thanks @iemejia !

**Comment by aviemzur** [comment]
@iemejia 
I actually found this log useful. Any reason why it was deleted? Perhaps the log can be returned with log level 'debug' instead of completely removing it?

**Pull request closed without merging**